pkg/controller/operators/catalog/operator.go

```diff
@@ -140,7 +145,7 @@ func (o *Operator) syncCatalogSources(obj interface{}) (syncError error) {
 
 	o.sourcesLock.Lock()
 	defer o.sourcesLock.Unlock()
-	o.sources[catsrc.GetName()] = src
+	o.sources[sourceKey{name: catsrc.GetName(), namespace: catsrc.GetNamespace()}] = src
```

Comment:
this is hard to review since I know most of this comes from the other PR. let's wait to merge that and then rebase this one.